src/main/java/com/roommatematcher/enterprise/service/ProfileServiceStub.java

```diff
@@ -24,10 +28,20 @@ public Profile fetchBySmokingPreference(Boolean smokingPreference) {
         return profile;
     }
 
+    //return array of all profiles and sift for desired location attribute
     @Override
-    public Profile fetchByLocation(String location) {
+    public ArrayList<Profile> fetchByLocation(String location) {
+        ArrayList<Profile> desiredProfiles = new ArrayList<Profile>();
         Profile profile = new Profile();
-        profile.setLocation("Dallas");
-        return profile;
+        desiredProfiles.add(profile);
+        //get all profiles and put them into an ArrayList
+        //desiredProfiles = ProfileDAOStub.fetchAll();
+        //.removeIf() is essentially a foreach with nested if statement:
+        // If e.location does not equal the location string
+        // that is passed in, we remove it from the ArrayList.
```

Comment:
Committing commented out code is considered bad practice in version control, because version control offers better alternatives:
1) Use a feature branch for experimental features.
2) Delete what is not needed.  You can always compare with history to see what used to be there.
Commented out code is considered clutter in a version control system.